### ISSUE TITLE
"Cast Shadow" with PCF3x3 fails to render dynamically generated mesh correctly

### ISSUE THREAD

**Issue by AbeLancs** [playcanvas/engine#4211]
Posting bug report as instructed from PlayCanvas forum: https://forum.playcanvas.com/t/light-shadow-bug/25343 

### Use of lights with "Cast Shadow" checked, and using PCF 3x3 shadow map breaks rendering of dynamically generated meshes in Windows/Android +(Firefox). Works fine on iOS (iPad Air).

Test project URL: https://playcanvas.com/editor/project/916295

### Steps to Reproduce

1. Check if both lights have "Cast Shadow" checkbox checked and one uses PCF 3x3 and other one has PCF 5x5.
2. When 5x5 is used, mesh is visible as expected, when 3x3 is used, texture alpha is respected but color is black.
3. Lightsources can be toggled by clicking the "Switch lightsource" option and console logs active lightsource.

Screen grabs:
![9d2f44bf67adae1db701b8bbfb6b563d9c9323d4_2_592x750](https://user-images.githubusercontent.com/42146181/164988735-36e5f625-f678-426f-99a9-678c8f6b963e.png)
![2378768da89abadcc9349abeebdfbef202af64bf_2_640x750](https://user-images.githubusercontent.com/42146181/164988736-3327a132-6ee9-4a9d-b88d-e1bada29160e.png)

**Comment by mvaligursky** [comment]
I cannot reproduce this on Mac Monterey 12.3.1 and Firefox 99.0.1 (64-bit), this seems like a Windows only issue at the moment (I have not tried to repro on Windows)

**Comment by Maksims** [comment]
Same problem here, Windows, RTX 3090, Chrome 100.0.4896.127

**Comment by slimbuck** [comment]
Managed to reproduce on window with:
ANGLE (Intel, Mesa Intel(R) UHD Graphics 630 (CFL GT2), OpenGL 4.6)

**Comment by AbeLancs** [comment]
@mvaligursky : Yes, I suspect Mac OS is not affected. iOS on iPad Air is not affected (with Safari). 
But it is definitely present on Windows 10/11 (64-bit)+Firefox  v99.0.1/Chrome v100.0.4896.127 and also Android 11+Firefox. Haven't tested on other configurations.
